### PR TITLE
Add hotspot rule S4797 (File Handling)

### LINKED ISSUE #1129: Rule S4797: Handling files is security-sensitive
Implements [RSPEC-4797](https://jira.sonarsource.com/browse/RSPEC-5076)

### PR DESCRIPTION
**Pull request by andrea-guarino-sonarsource**
Fixes #1129

### CODE REVIEW

File: eslint-bridge/src/rules/file-handling.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    callee.type === "MemberExpression" &&
+    callee.object.type === "Identifier" &&
+    isQuestionable(callee.property, args, getModuleNameOfIdentifier(callee.object, context))
+  ) {
+    context.report({ message: MESSAGE, node: callee });
+  } else if (
+    callee.type === "Identifier" &&
+    isQuestionable(callee, args, getModuleNameOfImportedIdentifier(callee, context))
+  ) {
+    context.report({ message: MESSAGE, node: callExpression });
```
Comment (vilchik-elena) [inline eslint-bridge/src/rules/file-handling.ts]: why do we report here on entire expression and above on callee only ?

File: eslint-bridge/src/rules/file-handling.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+function isQuestionable(
+  expression: estree.Expression,
+  args: Argument[],
+  moduleName?: estree.Literal,
+) {
+  if (!moduleName || moduleName.value !== FS_MODULE) {
+    return false;
+  }
+  if (expression.type === "Identifier") {
```
Comment (vilchik-elena) [inline eslint-bridge/src/rules/file-handling.ts]: it's always identifier, you can change parameter declared type

File: eslint-bridge/src/rules/file-handling.ts (hunk 1 above)
Comment (andrea-guarino-sonarsource) [reply eslint-bridge/src/rules/file-handling.ts]: Forgot that while refactoring, I will report only on callee

File: eslint-bridge/src/rules/file-handling.ts (hunk 2 above)
Comment (andrea-guarino-sonarsource) [reply eslint-bridge/src/rules/file-handling.ts]: As discussed, “MemberExpression” can have property that is not an identifier (like `fs[getFunctionName()]('foo.txt');`)

**Pull request merged**